ui_testing/testcases/basic_tests/test006_orders.py

```diff
@@ -3537,6 +3538,58 @@ def test100_create_multiple_suborders_with_testplans_testunits(self):
             else:
                 self.assertCountEqual(test_units_names, third_suborder_test_units)
 
+    def test101_create_order_with_test_plans_with_same_name(self):
+        """
+        Orders: Create Approach: Make sure In case you create two test plans with the same name
+        and different materiel type, the test units that belongs to them displayed correct in
+        analysis step two
+
+        LIMS-6296
+        """
+        test_plans_list = TestPlanAPI().create_double_completed_testplan_same_name_diff_material()
+        self.assertTrue(test_plans_list)
+        test_units_list = [tu['testUnits'][0]['name'] for tu in test_plans_list]
+        first_test_plan_dict = {'id': int(test_plans_list[0]['selectedTestPlan']['id']),
+                                'name': test_plans_list[0]['selectedTestPlan']['text'],
+                                'version': 1}
+        second_test_plan_dict = {'id': int(test_plans_list[1]['selectedTestPlan']['id']),
+                                 'name': test_plans_list[1]['selectedTestPlan']['text'],
+                                 'version': 1}
+        first_material = test_plans_list[0]['materialType'][0]
+        second_material = test_plans_list[1]['materialType'][0]
+        first_article = test_plans_list[0]['selectedArticles'][0]
+        second_article = test_plans_list[1]['selectedArticles'][0]
+        update_suborder = [{'testPlans': [first_test_plan_dict],
+                            'selectedTestPlans': [first_test_plan_dict],
+                            'materialType': first_material,
+                            'materialTypeId': first_material['id'],
+                            'article': first_article,
+                            'articleId': first_article['id']},
+                           {'testPlans': [second_test_plan_dict],
+                            'selectedTestPlans': [second_test_plan_dict],
+                            'materialType': second_material,
+                            'materialTypeId': second_material['id'],
+                            'article': second_article,
+                            'articleId': second_article['id']}]
```

Comment:
Create a method that takes the test_plan_list and returns the suborder_data

Reply:
done